### PR TITLE
Remove RouteGroup from RouteValues

### PR DESCRIPTION
**Pull request by ajaybhargavb**
#3250 
@rynowak @pranavkm 

Currently this does not break any tests. Will add tests once the fix is accepted.

### CODE REVIEW

**Comment by Eilon** [comment]
Talked to @ajaybhargavb , there might be a better fix for this.

**Comment by ajaybhargavb** [comment]
Updated with new fix and added a unit test.

File: src/Microsoft.AspNet.Mvc.Core/Infrastructure/MvcRouteHandler.cs [hunk 1]
```diff
@@ -66,6 +67,7 @@ public VirtualPathData GetVirtualPath(VirtualPathContext context)
             // without affecting something upstream.
             var oldRouteData = context.RouteData;
             var newRouteData = new RouteData(oldRouteData);
+            newRouteData.Values.Remove(AttributeRouting.RouteGroupKey);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/Infrastructure/MvcRouteHandler.cs]: Might want to group this with the next blok and make a omment that this is done to 'simulate' the output of onventional routing.

**Comment by rynowak** [comment]
:shipit:

**Pull request merged**